### PR TITLE
Azure Common Runtime unit tests and other consolidations

### PR DESCRIPTION
**Pull request by yugangw-msft**
Includes:
1. LRO unit test and related code changes for more readable code
2. Necessary new files so to wire up jshint, such as share ".jshintrc" with the generic runtime folder
3. Fix all jshint errors
4. Fix node js project files so that mocha test discovering works inside VS
5. Create a solution file to include both client runtime and azure runtime node.js projects, so global change will be easiler

### CODE REVIEW

**Comment by yugangw-msft** [comment]
CI failure was caused by the workaround we made to let node azure.common tests reference not yet published ms-rest package.

**Comment by yugangw-msft** [comment]
We will need to upgrade the npm version in the slave

File: ClientRuntimes/NodeJS/ClientRuntime.Azure/lib/azureServiceClient.js [hunk 1]
```diff
@@ -96,30 +97,30 @@ AzureServiceClient.prototype.getPutOperationResult = function (response, poller,
...
-  if (!response.response) {
-    return callback(new Error('Missing response.response'));
+  if (!resultOfInitialRequest.response) {
+    return callback(new Error('Missing resultOfInitialRequest.response'));
   }
   
-  if (response.response.statusCode != 200 &&
-      response.response.statusCode != 202 &&
-      response.response.statusCode != 204) {
+  if (resultOfInitialRequest.response.statusCode != 200 &&
```
Comment (amarzavery) [inline ClientRuntimes/NodeJS/ClientRuntime.Azure/lib/azureServiceClient.js]: !==

File: ClientRuntimes/NodeJS/ClientRuntime.Azure/lib/azureServiceClient.js [hunk 2]
```diff
@@ -25,31 +26,31 @@ util.inherits(AzureServiceClient, msrest.ServiceClient);
...
+  if (resultOfInitialRequest.response.statusCode !== 200 &&
+      resultOfInitialRequest.response.statusCode !== 201 &&
+      resultOfInitialRequest.response.statusCode !== 202) {
     return callback(new Error(util.format('Unexpected polling status code from long running operation \'%s\'', 
-      response.response.statusCode)));
+      resultOfInitialRequest.response.statusCode)));
   }
   
-  var pollingState = new PollingState(response, this.longRunningOperationRetryTimeout);
+  var pollingState = new PollingState(resultOfInitialRequest, this.longRunningOperationRetryTimeout);
```
Comment (amarzavery) [inline ClientRuntimes/NodeJS/ClientRuntime.Azure/lib/azureServiceClient.js]: this.longRunningOperationRetryTimeout needs to be defined inside the constructor

**Comment by yugangw-msft** [comment]
CI passes:http://azuresdkci.cloudapp.net/job/autorest-runtime/209/console

**Pull request merged**